integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -279,16 +276,23 @@ public int getattr(String path, FileStat stat) {
       stat.st_mtim.tv_sec.set(ctime_sec);
       stat.st_mtim.tv_nsec.set(ctime_nsec);
 
-      // for shell-based group mapping, use the uid and gid of the corresponding user registered in
-      // unix; otherwise use uid and gid of the user running alluxio-fuse
+      // for shell-based group mapping, try to get the uid and gid of file/dir owner.
+      // If the shell-based group mapping is not enabled in Fuse
+      // or the user does not registered in the unix,
+      // uid and gid of the user running alluxio-fuse will be set.
+      long uid = -1;
+      long gid = -1;
       if (mIsShellGroupMapping) {
         String owner = status.getOwner();
-        stat.st_uid.set(AlluxioFuseUtils.getUid(owner));
-        stat.st_gid.set(AlluxioFuseUtils.getGid(owner));
-      } else {
-        stat.st_uid.set(UID);
-        stat.st_gid.set(GID);
+        uid = AlluxioFuseUtils.getUid(owner);
+        gid = AlluxioFuseUtils.getGid(owner);
```

Comment:
is `gid = AlluxioFuseUtils.getGid(owner);` actually correct? Even if the user/group in Alluxio is lu/lugroup, I think this might map to something like lu/staff

Reply:
Two circumstances may happen:
1) `lugroup` is one of the groups of `lu` user, but `staff` is the primary group of `lu`.
In this case, we would be better to use original `lu/lugroup`. It is a place that I need to fix.
2) in another computer, another user `lu` with group `lugroup` writes the file. my computer also has user `lu` but does not have group `lugroup`. In this case, I think `lu/staff` is fine.

So maybe we should first `getUid(owner)` and `getGid(group)`.
If `getUid(owner)` failed, we use the user group of the user mounting the Alluxio-Fuse
If `getUid(owner)` passed but `getGid(group)` failed, we use the current `AlluxioFuseUtils.getGid(owner)` to get the group of current user.
If both passed, we check if the user belongs to the group in the current unix.

Good catch! Thanks a lot

Reply:
sg, but we should be careful about (2) - if `lugroup` is only for 3 people but `staff` contains everyone in the organization, converting `lugroup` to `staff` could give everyone access to the file, when only `lu` and people in `lugroup` are supposed to have access. This could get us in trouble

Reply:
The current implementation of `uid/gid` is only used in  `getAtttr` (the ls command) to show to users, but it does not actually change the user/group of the files. `chown`, `write` and `read` still refer to the original permission.

I should document this part.

Reply:
I see, so it isn't a security risk. In that case I suggest using nobody/nouser if the uid can't be resolved from the username, and nogroup if the gid can't be resolved from the group name. Agree that it's better to show everything instead of throwing an exception

Reply:
Agree! Change to use nobody(uid=-1) and nogroup(gid=-1) if uid or gid cannot be resolved.
Thanks